sdk/core/System.ClientModel/src/ModelReaderWriter/SequenceBufferReader.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Buffers;
+using System.Diagnostics;
+
+namespace System.ClientModel.Internal;
+
+internal class SequenceBufferReader : IDisposable
```

Comment:
Could we call it ReadOnlySequence?

Reply:
Are we concerned at all with naming conflict on `ReadOnlySequence<T>`?

Reply:
We had an offline discussion. We settled on BufferSequence, BufferSequence.Reader, and BufferSegment